waspc/src/Wasp/Psl/Ast/Schema.hs

```diff
@@ -49,3 +51,6 @@ getGenerators schema = [generator | generator@((Psl.ConfigBlock.ConfigBlock Psl.
 
 getConfigBlocks :: Schema -> [ConfigBlock]
 getConfigBlocks (Schema blocks) = [configBlock | ConfigBlock configBlock <- blocks]
+
+getModelNames :: Schema -> [String]
+getModelNames schema = map Model.getName $ getModels schema
```

Comment:
There's always `Wasp.Psl.Util` which seems like an okay place to put this helper.

Reply:
Well it is not so terrible, but I get what you mean: I mean it is quite random, why can't caller just do `Model.getName <$> getModels schema`? I don't see much value in decoupling caller from schema's internals -> the fact that schema has models in it are not internals, nor that each model has a name, it is basically its API.
The fact that you have a caller which needs just model names but doesn't care how exactly to fetch models or a name from a model doesn't seem like a good enough reason to create a whole new API here for that use case.

So I would say, either move this function to wherever you need it (so much close to the caller and out of Wasp/Psl module subtree), or even better, get rid of this function and just inline it wherever you are using it -> again, I don't think you gain anyting by abstracting stuff here.

Reply:
> `Model.getName <$> getModels schema`

This is the part I'm not a fan of. We do that often (inline accesses to source-of-truth objects), but it always bugs me a little.

My main motivation here is [Law of Demeter](https://en.wikipedia.org/wiki/Law_of_Demeter).

The counter-argument (if I got it correctly) is that, in this case, the schema's internal structure is so inherent that there's no point in hiding it (and that's why this particular instance is not the best example, but bear with me).

I generally dislike chaining multiple "accessor" calls. It requires importing multiple symbols, and refactoring many imports and call sites whenever I make a simple name change.

I'm talking about the stuff like:
```haskell
import Foo
import Bar
import AppSpec

... Foo.getBar <$> Foo.getBars $ AS.getFoo spec

-- then later again

... Foo.getBar <$> Foo.getBars $ AS.getFoo spec
```
Sometimes someone notices the repetition and defines a local helper function (like you suggest here). Then, in a different location, someone else notices the repetition again and defines an identical helper function (again local). We end up with N inline queries and M local functions that all do the same thing.

For example, this `getModelNames` function is used in two modules: `Wasp.Project.Analyze` and `Wasp.LSP.Prisma`.

Ideally, If I want something from the AppSpec in the generator, I don't want to import the entire hierarchy and inline the same expression multiple times.
I want to import only a single function that gives me what I want and then use that.

In simple terms, I'd prefer if our source-of-truth objects (like the AppSpec and the schema) had a higher-level API for stuff that's often used. It would be perfect if we could use an index file to reexport stuff we want to expose and "ban" all lower-level imports (but I don't think Haskell can do this).

Reply:
I'm leaving this one for later as well.

Reply:
Done in #2410